core/src/test/java/io/grpc/internal/DnsNameResolverTest.java

```diff
@@ -177,6 +180,8 @@ public void invalidDnsName_containsUnderscore() {
 
   @Test
   public void resolve() throws Exception {
+    // disable network address cache
+    Security.setProperty(DnsNameResolver.NETWORKADDRESS_CACHE_TTL_SECURITY_PROPERTY, "-1");
```

Comment:
Any global changes should be restored by the end of the test. Either a try-catch or a `@Before` to save the value before the test runs and a `@After` to restore it.